src/compas_fab/ghpython/components/plan_cartesian_motion.py

```diff
@@ -0,0 +1,114 @@
+        self.SetUpParam(p, "start_configuration", "start_configuration",
+                        "The robot's full configuration, i.e. values for all configurable joints of the entire robot, at the starting position. Defaults to the all-zero configuration.")  # noqa E501
+        p.Access = Grasshopper.Kernel.GH_ParamAccess.item
+        self.Params.Input.Add(p)
+
+        p = Grasshopper.Kernel.Parameters.Param_String()
+        self.SetUpParam(p, "group", "group", "The planning group used for calculation. Defaults to the robot's main planning group.")
+        p.Access = Grasshopper.Kernel.GH_ParamAccess.item
+        self.Params.Input.Add(p)
+
+        p = Grasshopper.Kernel.Parameters.Param_GenericObject()
+        self.SetUpParam(p, "attached_collision_meshes", "attached_collision_meshes", "A list of attached collision meshes to be included for planning.")
+        p.Access = Grasshopper.Kernel.GH_ParamAccess.list
+        self.Params.Input.Add(p)
+
+        p = Grasshopper.Kernel.Parameters.Param_GenericObject()
+        self.SetUpParam(p, "path_constraints", "path_constraints",
+                        "Optional constraints that can be imposed along the solution path. Note that path calculation won't work if the start_configuration violates these constraints. Defaults to None.")  # noqa E501
+        p.Access = Grasshopper.Kernel.GH_ParamAccess.list
+        self.Params.Input.Add(p)
+
+        p = Grasshopper.Kernel.Parameters.Param_Number()
+        self.SetUpParam(p, "max_step", "max_step", "The approximate distance between the calculated points. (Defined in the robot's units.) Defaults to 0.01.")
+        p.Access = Grasshopper.Kernel.GH_ParamAccess.item
+        self.Params.Input.Add(p)
+
+        p = Grasshopper.Kernel.Parameters.Param_Boolean()
+        self.SetUpParam(p, "compute", "compute", "If `True`, calculates a trajectory.")
+        p.Access = Grasshopper.Kernel.GH_ParamAccess.item
+        self.Params.Input.Add(p)
+
+    def RegisterOutputParams(self, pManager):
+        p = Grasshopper.Kernel.Parameters.Param_GenericObject()
+        self.SetUpParam(p, "trajectory", "trajectory", "The calculated trajectory.")
+        self.Params.Output.Add(p)
+
+    def SolveInstance(self, DA):
+        p0 = self.marshal.GetInput(DA, 0)
+        p1 = self.marshal.GetInput(DA, 1)
+        p2 = self.marshal.GetInput(DA, 2)
+        p3 = self.marshal.GetInput(DA, 3)
+        p4 = self.marshal.GetInput(DA, 4)
+        p5 = self.marshal.GetInput(DA, 5)
+        p6 = self.marshal.GetInput(DA, 6)
+        p7 = self.marshal.GetInput(DA, 7)
+        result = self.RunScript(p0, p1, p2, p3, p4, p5, p6, p7)
+
+        if result is not None:
+            self.marshal.SetOutput(result, DA, 0, True)
+
+    def get_Internal_Icon_24x24(self):
+        return plan_cartesian_motion_icon
+
+    def RunScript(self, robot, planes, start_configuration, group, attached_collision_meshes, path_constraints, max_step, compute):
```

Comment:
While I know we do this in ghpython components, is the idea of having a `bool` to trigger long computation a common practice in other GH components?

Reply:
i don't know, but like that you are somewhat in control if you know that this component takes a bit long.

Reply:
👍